dynamiqs/utils/optimal_control.py

```diff
@@ -1,103 +1,72 @@
         seed: Seed for the random number generator.
-        dtype: Real data type of the returned tensor.
-        device: Device of the returned tensor.
+        dtype: Complex data type of the returned array.
 
     Returns:
-        _(*size)_ Tensor filled with random real numbers.
+        _(array of shape (*shape))_ Array filled with random real numbers.
 
     Examples:
-        >>> x = dq.rand_real((2, 5), max=5.0, seed=42)
-        >>> x
-        tensor([[ 3.823,  4.150, -1.171,  4.593, -1.096],
-                [ 1.009, -2.434,  2.936,  4.408, -3.668]])
+        >>> dq.rand_real((2, 5), max=5.0, seed=42)
+        Array([[3.22 , 1.613, 0.967, 4.432, 4.21 ],
+               [0.96 , 1.726, 1.262, 3.16 , 3.274]], dtype=float32)
     """
-    # Note: We need to manually fetch the default device, because if `device` is `None`
-    # `torch.Generator` picks "cpu" as the default device, and not the device set by
-    # `torch.set_default_device`.
-    device = to_device(device)
-
-    # define random number generator from seed
-    generator = torch.Generator(device=device)
-    generator.seed() if seed is None else generator.manual_seed(seed)
+    dtype = get_rdtype(dtype)
+    shape = (shape,) if isinstance(shape, int) else shape
 
-    rdtype = get_rdtype(dtype)
+    # define PRNG key from seed
+    key = jax.random.PRNGKey(seed)
 
-    # generate random real with values in [-max, max[
-    rand_01 = torch.rand(size, generator=generator, dtype=rdtype, device=device)
-    x = max * (rand_01 * 2 - 1)
-
-    # start tracking operations on the tensor if requires_grad is True
-    x.requires_grad_(requires_grad)
-
-    return x
+    # sample uniformly in [min, max)
+    return jax.random.uniform(key, shape=shape, dtype=dtype, minval=min, maxval=max)
 
 
 def rand_complex(
-    size: int | tuple[int, ...],
+    shape: int | tuple[int, ...],
     *,
     rmax: float = 1.0,
-    requires_grad: bool = False,
-    seed: int | None = None,
-    dtype: torch.complex64 | torch.complex128 | None = None,
-    device: str | torch.device | None = None,
-) -> Tensor:
-    r"""Returns a tensor filled with random complex numbers uniformly distributed in
+    seed: int = 0,
+    dtype: jnp.complex64 | jnp.complex128 | None = None,
+) -> Array:
+    r"""Returns an array filled with random complex numbers uniformly distributed in
     the complex plane.
 
-    Each element of the returned tensor has a random magnitude in the interval
-    $[0, \texttt{rmax})$ and a random phase in the interval $[0, 2\pi)$. Formally, each
-    element is defined by
-    $$
-        x = re^{i\theta}\ \text{with}\
-        \begin{cases}
-            r = \texttt{rmax} \cdot \sqrt{\texttt{rand(0,1)}} \\\\
-            \theta = 2\pi \cdot \texttt{rand(0,1)}
-        \end{cases}
-    $$
-    where $\texttt{rand(0,1)}$ is a random number uniformly distributed between 0 and 1.
+    Each element of the returned array is sampled uniformly in the disc of radius
```

Comment:
```suggestion
    Each element of the returned array is sampled uniformly in the disk of radius
```
(disc also exists, but you use disk below).